src/lib/packager/clients/repositories.rb

```diff
@@ -1791,7 +1791,12 @@ def repo_replace_handler(source_state, global_current)
       old_url = url2
       plaindir = Ops.get_string(generalData, "type", "YaST") == @plaindir_type
 
-      SourceDialogs.SetRepoName(source_state.fetch("raw_name", ""))
+      repo_name = source_state.fetch("raw_name", "")
+      # if the repository does not define a name then "raw_name" is empty,
+      # display "name" which contains the repository alias in that case
+      repo_name = source_state.fetch("name", "") if repo_name.empty?
```

Comment:
Is possible to have a test case or is it too hard?

Reply:
Unfortunately this is an old YCP code with a [500+lines method](https://github.com/yast/yast-packager/blob/067d1abb029ef154bb00d904f696da20ce6fdfc3/src/lib/packager/clients/repositories.rb#L847-L1394).... :worried: